common/src/main/java/com/microsoft/identity/common/internal/fido/LegacyFido2ApiManager.kt

```diff
@@ -0,0 +1,142 @@
+                                      allowedCredentials: List<String>?,
+                                      userVerificationPolicy: String,
+                                      span: Span
+    ): String = suspendCancellableCoroutine { continuation ->
+        val methodTag = "$TAG:authenticate"
+        val canceller = CancellationSignal()
+        continuation.invokeOnCancellation { canceller.cancel() }
+        span.setAttribute(
+            AttributeName.fido_manager.name,
+            TAG
+        )
+        val requestOptions = com.google.android.gms.fido.fido2.api.common.PublicKeyCredentialRequestOptions.Builder()
+            .setChallenge(challenge.toByteArray(Charsets.UTF_8))
+            .setRpId(relyingPartyIdentifier)
+            .build()
+        Logger.info(methodTag, "Calling the legacy FIDO2 API with public key credential options to get a PendingIntent.")
+        val result = legacyApi.getSignPendingIntent(requestOptions)
+
+        result.addOnSuccessListener(OnSuccessListener { pendingIntent ->
+            if (pendingIntent != null) {
+                Logger.info(methodTag, "Launching the legacy FIDO2 API PendingIntent.")
+                fragment.fidoLauncher?.launch(
+                    LegacyFido2ApiObject(
+                        assertionCallback = { assertion ->
+                            if (continuation.isActive) {
+                                continuation.resume(assertion)
+                            }
+                        },
+                        errorCallback = { exception ->
+                            if (continuation.isActive) {
+                                continuation.resumeWithException(exception)
+                            }
+                        },
+                        pendingIntent = pendingIntent
+                    ))
+                if (fragment.fidoLauncher == null) {
+                    val e = LegacyFido2ApiException(
+                        LegacyFido2ApiException.NULL_OBJECT,
+                        "fidoLauncher is null, which indicates that the legacy FIDO2 API is being used where it shouldn't be."
+                    )
+                    Logger.error(methodTag, e.message, e)
+                    if (continuation.isActive) {
+                        continuation.resumeWithException(e)
+                    }
+                }
+            } else {
+                val e = LegacyFido2ApiException(
+                    LegacyFido2ApiException.NULL_OBJECT,
+                    "Returned PendingIntent from legacy API is null."
+                )
+                Logger.error(methodTag, e.message, e)
+                if (continuation.isActive) {
+                    continuation.resumeWithException(e)
+                }
+            }
+        })
+        result.addOnFailureListener(OnFailureListener { exception ->
+            Logger.error(methodTag, "Failed to get a PendingIntent from the legacy FIDO2 API.", exception)
```

Comment:
nit: maybe this can be extracted into a helper function

```Kotlin
fun logAndReturnError(val continuation, val error){
   Logger.error(methodTag, error.message, e)
   if (continuation.isActive) {
      continuation.resumeWithException(error)
   }
}
```

Reply:
(also... for my knowledge. Is there any particular reason you surround every .resume() with isActive check? 

Did you run into any errors without it? or is this just added as a precaution?

I never use this class, and the sample doesn't seem to have that. (hence the Q)

Reply:
Extracted into a helper method as suggested.
The isActive checks are added as a precaution since the logic lies in callbacks. It hopefully will protect against any possibility in which the coroutine gets cancelled, but the Intent is active before then, and once it finishes it calls the callbacks.